server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -1159,18 +1138,33 @@ private Iterable<List<StoreFileMetadata>> getGroupedFilesIterable() {
          * <p>
          * NOTE: this diff will not contain the {@code segments.gen} file. This file is omitted on recovery.
          */
-        public RecoveryDiff recoveryDiff(MetadataSnapshot recoveryTargetSnapshot) {
+        public RecoveryDiff recoveryDiff(MetadataSnapshot recoveryTargetSnapshot, boolean ignoreMissingFiles) {
```

Comment:
Pls update javadoc explaining this param.

The flag is changing the behavior outlined in the javadoc of a stronger cksum where any missing/different file will cause identical files to be re-fetched.  With a `ignoreMissingFiles` value of true that is not the case.